sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -596,14 +596,17 @@ object CollapseProject extends Rule[LogicalPlan] {
 object CollapseRepartition extends Rule[LogicalPlan] {
```

Comment:
Please also add new test cases to `CollapseRepartitionSuite` for the changes in this rule.